site/docs/plugins/session.md

```diff
@@ -152,23 +147,136 @@ bot.start();
 Note how we also have to [adjust the context type](/guide/context.md#customising-the-context-object) to make the session available on it.
 The context flavor is called `SessionFlavor`.
 
-It is important (but optional) that you specify the `initial` option for the session middleware.
+### Initial Session Data
+
+When a user first contacts your bot, no session data is available for them.
+It is therefore important that you specify the `initial` option for the session middleware.
 Pass a function that generates a new object with initial session data for new chats.
 
+```ts
+/** Creates a new object that will be used as initial session data */
+function createInitialSessionData() {
+  return {
+    pizzaCount: 0,
+    // more data here
+  };
+}
+bot.use(session({ initial: createInitialSessionData }));
+```
+
+Same but much shorter:
+
+```ts
+bot.use(session({ initial: () => ({ pizzaCount: 0 }) }));
+```
+
+::: warning Sharing objects
+Make sure to always create a _new object_.
+Do **NOT** do this:
+
+```ts
+// DANGER, BAD, WRONG, STOP
+bot.use(session({ initial: { pizzaCount: 0 } })); // EVIL
+```
+
+If you would do this, several chats might share the same session object in memory.
+Hence, changing the session data in one chat may accidentally impact the session data in the other chat.
+:::
+
+You may also omit the `initial` option entirely, even though you are well advised not to do that.
+If you don't specify it, reading `ctx.session` will throw an error for new users.
+
+### Session Keys
+
 You can specify which session key to use by passing a function called `getSessionKey` to the [options](https://doc.deno.land/https/deno.land/x/grammy/mod.ts#SessionOptions).
-Whenever `getSessionKey` returns `undefined`, `ctx.session` will be `undefined`.
+That way, you can fundamentally change the way how the session plugin works.
+By default, data is stored per chat.
+Using `getSessionKey` allows you to store data per user, or per user-chat combination, or however you want.
+Here are three examples:
 
-In the example above, the session data is stored in your RAM, so as soon as your bot is stopped, all data is lost.
-This is convenient when you develop your bot or if you run automatic tests (no database setup needed), however, **that is most likely not desired in production**.
-In production, you should use the `storage` option of the session middleware to connect it to your datastore.
-There may already be storage adapter written for grammY that you can use (see below), but if not, it usually only takes 5 lines of code to implement one yourself.
+<CodeGroup>
+ <CodeGroupItem title="TypeScript" active>
+
+```ts
+/** stores data per chat (default) */
```

Comment:
It seems we should use `// some comments` to add comments?
In #104 we have already change all `/** */` to `//`, it will be better we use the same style.
```suggestion
// Stores data per chat (default).
```

Reply:
Good point.